pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -1149,10 +1178,30 @@ func updateStorageUsage(vm *vmopv1.VirtualMachine, moVM mo.VirtualMachine) {
 		vm.Status.Storage.Used.Other = kubeutil.BytesToResource(other)
 	}
 
+	if vmSnapshotUsed > 0 || volSnapshotUsed > 0 {
```

Comment:
I feel like this entire if block would be more clear as a switch statement. It also maps to Effective Go per https://go.dev/doc/effective_go#switch